commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareFormsEventParser.java

```diff
@@ -54,7 +54,12 @@ public String parseEventSubject(String eventSubject, Map<String, Object> eventPa
 
         if (eventSubject.equals(FORMS_EVENT)) {
             String xmlns = (String) ((Map) eventParameters.get(ATTRIBUTES)).get("xmlns");
-            return eventSubject.concat(".").concat(configName).concat(".").concat(xmlns);
+            String appId = (String) ((Map) eventParameters.get(ATTRIBUTES)).get("app_id");
+            if (appId != null) {
+                return eventSubject.concat(".").concat(configName).concat(".").concat(xmlns).concat(appId);
+            } else {
+                return eventSubject.concat(".").concat(configName).concat(".").concat(xmlns);
+            }
```

Comment:
there should be one return statement, please also simplify this code. Additionally add warning that app_id is null